src/platform/site-wide/mega-menu/containers/Main.jsx

```diff
@@ -25,12 +25,24 @@ export function flagCurrentPageInTopLevelLinks(
   href = window.location.href,
   pathName = window.location.pathname,
 ) {
-  return links.map(
-    link =>
-      pathName.endsWith(link.href) || href.includes(link.href)
-        ? { ...link, currentPage: true }
-        : link,
-  );
+  // Removing the trailing slash allows for a page's URL to have it or not.
+  const noTrailingSlashPathName = pathName.endsWith('/')
+    ? pathName.slice(0, -1)
+    : pathName;
+  return links.map(link => {
+    let processedLink = link;
+    if (link.href !== undefined) {
+      const noTrailingSlashHref = link.href.endsWith('/')
+        ? link.href.slice(0, -1)
+        : link.href;
+      processedLink =
+        noTrailingSlashPathName.endsWith(noTrailingSlashHref) ||
```

Comment:
I"m confused about this piece, and wonder if it is flipped around. When would a pathName end with an entire href? 

If we had a url like `""https://www.va.gov/foo/"`, then `noTrailingSlashHref` would be `"https://www.va.gov/foo"` and `noTrailingSlashPathName` would be `"/foo"`, so I don't think this would ever be true?

Or do I have Friday brain?

Reply:
Yeah, seeing `link.href` in proximity to `window.location.href` confused me. Naming things is hard, etc.

Thanks for clarifying!